mmcv/ops/points_sampler.py

```diff
@@ -0,0 +1,183 @@
+from typing import List
+
+import torch
+from torch import nn as nn
+
+from mmcv.runner import force_fp32
+from .furthest_point_sample import (furthest_point_sample,
+                                    furthest_point_sample_with_dist)
+
+
+def calc_square_dist(point_feat_a, point_feat_b, norm=True):
+    """Calculating square distance between a and b.
+
+    Args:
+        point_feat_a (Tensor): (B, N, C) Feature vector of each point.
+        point_feat_b (Tensor): (B, M, C) Feature vector of each point.
+        norm (Bool, optional): Whether to normalize the distance.
+            Default: True.
+
+    Returns:
+        Tensor: (B, N, M) Distance between each pair points.
+    """
+    num_channel = point_feat_a.shape[-1]
+    # [bs, n, 1]
+    a_square = torch.sum(point_feat_a.unsqueeze(dim=2).pow(2), dim=-1)
+    # [bs, 1, m]
+    b_square = torch.sum(point_feat_b.unsqueeze(dim=1).pow(2), dim=-1)
+
+    corr_matrix = torch.matmul(point_feat_a, point_feat_b.transpose(1, 2))
+
+    dist = a_square + b_square - 2 * corr_matrix
+    if norm:
+        dist = torch.sqrt(dist) / num_channel
+    return dist
+
+
+def get_sampler_type(sampler_type):
+    """Get the type and mode of points sampler.
+
+    Args:
+        sampler_type (str): The type of points sampler.
+            The valid value are "D-FPS", "F-FPS", or "FS".
+
+    Returns:
+        class: Points sampler type.
+    """
+    if sampler_type == 'D-FPS':
+        sampler = DFPS_Sampler
+    elif sampler_type == 'F-FPS':
+        sampler = FFPS_Sampler
+    elif sampler_type == 'FS':
+        sampler = FS_Sampler
+    else:
+        raise ValueError('Only "sampler_type" of "D-FPS", "F-FPS", or "FS"'
+                         f' are supported, got {sampler_type}')
+
+    return sampler
+
+
+class Points_Sampler(nn.Module):
```

Comment:
should we rename the `Points_Sampler` to `PointsSampler`

Reply:
Since our main goal is to be consistent with mmdet3d, maybe we should keep the original name?  @ZwwWayne

Reply:
Use PointsSampler